### PR TITLE
feat: kong gateway's aws vault backend now supports automatic IAM credential fetch and role assuming

### PR DESCRIPTION
**Pull request by windmgc**
KAG-1542

This is a combination update of KAG-1991 and KAG-1993

This is an enhancement of Kong Gateway's Vault AWS backend, to let it support multiple ways of automatic IAM credential fetching, and role assuming.

### Testing instructions

Preview link: 

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

**Comment by windmgc** [comment]
Hmm, the smoke test seems to be broken for a long time(at least since the last time I raised a PR here)

is there anyway I can get a preview for this PR?

File: app/_src/gateway/kong-enterprise/secrets-management/backends/aws-sm.md [hunk 1]
```diff
@@ -138,9 +158,12 @@ Configuration options for an AWS Secrets Manager vault in {{site.base_gateway}}:
 Parameter | Field name                     | Description
 ----------|--------------------------------|------------
 `vaults.config.region` | **AWS region** | The AWS region your vault is located in.
+`vaults.config.endpoint_url` | **AWS Secrets Manager Endpoint URL** | The endpoint URL of AWS Secrets Manager service.If not specified, the value used by vault will be the official AWS SecretsManager service url which is `https://secretsmanager.<region>.amazonaws.com`. You can specify a complete URL(including the "http/https" scheme) to override the endpoint that vault will connect to.
+`vaults.config.assume_role_arn` | **Assume AWS IAM role ARN** | The target IAM role ARN that will be assumed to call AWS Secrets Manager service. If specified, Vault backend will do additional role assuming based on your current runtime's IAM Role. If you are not using assume role, you should not specify this value.
```
Comment (hanshuebner) [inline app/_src/gateway/kong-enterprise/secrets-management/backends/aws-sm.md:162]:
```suggestion
`vaults.config.assume_role_arn` | **Assume AWS IAM role ARN** | The target IAM role ARN that will be assumed to call AWS Secrets Manager service. If specified, the Vault backend will do additional role assuming based on your current runtime's IAM Role. If you are not using assume role, do not specify this value.
```

**Review by hanshuebner — CHANGES_REQUESTED** [review]
Just a couple of grammar changes, looks good to me otherwise.

**Review by hanshuebner — CHANGES_REQUESTED** [review]
Just a couple of grammar changes, looks good to me otherwise.

**Comment by windmgc** [comment]
@hanshuebner Done! Thanks for the suggestions!

**Review by zhongweiy — APPROVED** [review]

**Review by hanshuebner — APPROVED** [review]

**Review by Guaris — COMMENTED** [review]
lgtm

**Pull request merged**